src/components/ComputeModal.js

```diff
@@ -953,8 +952,8 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
                     },
                     options: [
                       { value: false, label: 'Standard VM', isDisabled: requiresSpark },
-                      { value: 'master', label: 'Spark master node' },
-                      { value: 'cluster', label: 'Spark cluster' }
+                      { value: 'master', label: 'Spark master node', isDisabled: isRStudio },
```

Comment:
Hmm... After thinking about it a bit more, I don't think I like having the image itself have a flag for `isRStudio`. 

This isn't a scalable approach if, say in the future we need to disable/enable options for Jupyter images. I think a better approach is to leverage this [tools dictionary](https://github.com/DataBiosphere/terra-ui/blob/dev/src/components/notebook-utils.js#L75), and check if the `selectedLeoImage` id is present in the `imageIds` array. This approach is more flexible too, as it doesn't require any terra-docker changes and building of the .json file for re-classification of images. 

I would prefer to use this tools dictionary for all current uses of `isRStudio` I think. What are your thoughts? cc @gpcarr

Reply:
I agree that it would be great to expose this via a property/method in the tools dictionary instead of sprinkling isRStudio around!